bundle/phases/deploy.go

```diff
@@ -319,6 +319,22 @@ func logTelemetry(ctx context.Context, b *bundle.Bundle) {
 		bundleUuid = b.Config.Bundle.Uuid
 	}
 
+	variableCount := len(b.Config.Variables)
+	complexVariableCount := int64(0)
+	lookupVariableCount := int64(0)
+	for _, v := range b.Config.Variables {
+		// If the resolved value of the variable is a complex type, we count it as a complex variable.
+		// We can't rely on the "type: complex" annotation because the annotation is optional in some contexts
```

Comment:
but if you only have annotation and nothing else (no value -- it's unused variable), then you have to rely on annotation, right?

Reply:
We don't allow variables without a value:
https://github.com/databricks/cli/blob/f6b0b4615047682a821f0b537817208ebfcfb4eb/bundle/config/mutator/set_variables.go#L94

It is possible that users might annotate a variable as a complex type but use it as primitive, in which case we would rightly not track it.